apps/sports-helsinki/src/domain/event/eventKeywords/EventKeywords.tsx

```diff
@@ -76,82 +77,92 @@ const EventKeywords: React.FC<Props> = ({
-          onClick={withActions ? handleClick('isFree') : undefined}
-        />
-      )}
-      {showKeywords &&
-        first &&
-        (showKeywordsCount ? customTagsCount < 2 : true) && (
+    <div className={styles.keywordsWrapper}>
+      <div className={styles.keywordsMain}>
+        {today && (
           <KeywordTag
             color="engelLight50"
-            blackOnMobile={blackOnMobile}
-            hideOnMobile={hideKeywordsOnMobile}
-            key={first.id}
-            keyword={first.name}
-            onClick={withActions ? handleClick('text', first.name) : undefined}
+            keyword={t('categories.labelToday')}
+            onClick={
+              withActions
+                ? handleClick('dateType', DATE_TYPES.TODAY)
+                : undefined
+            }
           />
         )}
-      {showKeywords &&
-        second &&
-        (showKeywordsCount
-          ? customTagsCount + Number(Boolean(first)) < 2
-          : true) && (
+        {!today && thisWeek && (
           <KeywordTag
             color="engelLight50"
-            blackOnMobile={blackOnMobile}
-            hideOnMobile={hideKeywordsOnMobile}
-            key={second.id}
-            keyword={second.name}
-            onClick={withActions ? handleClick('text', second.name) : undefined}
+            keyword={t('categories.labelThisWeek')}
+            onClick={
+              withActions
+                ? handleClick('dateType', DATE_TYPES.THIS_WEEK)
+                : undefined
+            }
           />
         )}
-      {!!restKeywords.length &&
-        showKeywords &&
-        !showKeywordsCount &&
-        restKeywords.map((tag) => (
+        {showIsFree && freeEvent && (
           <KeywordTag
-            color="engelLight50"
+            color="tramLight20"
+            keyword={t('categories.labelFree')}
+            onClick={withActions ? handleClick('isFree') : undefined}
+          />
+        )}
+        {showKeywords &&
+          first &&
+          (showKeywordsCount ? customTagsCount < 2 : true) && (
+            <KeywordTag
+              color="engelLight50"
+              blackOnMobile={blackOnMobile}
+              hideOnMobile={hideKeywordsOnMobile}
+              key={first.id}
+              keyword={first.name}
+              onClick={
+                withActions ? handleClick('text', first.name) : undefined
+              }
+            />
+          )}
+        {showKeywords &&
+          second &&
+          (showKeywordsCount
+            ? customTagsCount + Number(Boolean(first)) < 2
+            : true) && (
+            <KeywordTag
+              color="engelLight50"
+              blackOnMobile={blackOnMobile}
+              hideOnMobile={hideKeywordsOnMobile}
+              key={second.id}
+              keyword={second.name}
+              onClick={
+                withActions ? handleClick('text', second.name) : undefined
+              }
+            />
+          )}
```

Comment:
I'm again confused with this - like I think I was earlier too. Would it make more sense to collect a base component that is shared in every condition (or at least in many) and then just override the values that changes?

```
const base = {
color: "engelLight50",
blackOnMobile: blackOnMobile,
// ...
}

<KeywordTag {...base} color='overriddencolor' key={second.id} keyword={second.name} />
```

Reply:
Maybe an extending KeywordTag in the same file with some fixed props would help?

```
      {today && (
        <KeywordTag
          color="engelLight50"
          keyword={t('categories.labelToday')}
          onClick={
            withActions ? handleClick('dateType', DATE_TYPES.TODAY) : undefined
          }
        />
      )}
```

would become 

```
  const TodayKeyword = () => (
    <KeywordTag
      color="engelLight50"
      keyword={t('categories.labelToday')}
      onClick={
        withActions ? handleClick('dateType', DATE_TYPES.TODAY) : undefined
      }
    />
  );
  ``` 
  
  etc